bluesky/run_engine.py

```diff
@@ -1212,12 +1309,21 @@ def _stage(self, msg):
 
     @asyncio.coroutine
     def _unstage(self, msg):
+        """Instruct the RunEngine to unstage the object
+
+        Expected message object is:
+
+            Msg('unstage', object)
+        """
         _, obj, args, kwargs = msg
         # If an object has no 'unstage' method, assume there is nothing to do.
         if not hasattr(obj, 'unstage'):
             return
         result = obj.unstage()
```

Comment:
Good catch with this issue -- a win for your fuzz testing efforts.

In ophyd, the rule with staging/unstaging is: If an object is staged, you have to unstage it before you can stage it again. But if an object is unstaged, it's OK to unstage it again (nothing will happen). To reflect this, I think change `remove` to `discard`, which does nothing if the element to be discarded is not present.